src/device.rs

```diff
@@ -85,9 +90,12 @@ pub unsafe extern "C" fn wgpuAdapterGetProperties(
     let maybe_props = gfx_select!(adapter => GLOBAL.adapter_get_info(adapter));
     match maybe_props {
         Ok(props) => {
-            //let name = std::ffi::CString::new(props.name);
-            //store_object_globally(name); --> how?
-            //properties.name = c_string.unwrap(name).as_ptr();
+            properties.name = ADAPTER_NAMES
+                .lock()
+                .unwrap()
+                .entry(adapter)
+                .or_insert_with(|| CString::new((&props.name) as &str).unwrap())
```

Comment:
why do we need `(&props.name) as &str` as opposed to just `&props.name`?

Reply:
With ```&props.name```, the compiler complains about an unsatisfied trait bound: ```the trait bound `Vec<u8>: From<&std::string::String>` is not satisfied```. With ```(&props.name) as &str```, it works fine.

Reply:
`props.name.as_str()` would be even better :)